src/backgroundblurprocessor/BackgroundBlurProcessorProvided.ts

```diff
@@ -415,3 +449,33 @@ export default class BackgroundBlurProcessorProvided implements BackgroundBlurPr
     return supportsBlurFilter;
   }
 }
+
+/** @internal */
+interface MonitorCpuObserver {
+  reduceCPUUtilization: () => void;
+  increaseCPUUtilization: () => void;
+}
+
+/** @internal */
+class BackgroundFilterMonitor implements BackgroundBlurVideoFrameProcessorObserver {
+  private lastEventTimestamp: number = 0;
+  constructor(private monitoringPeriodMillis: number, private observer: MonitorCpuObserver) {}
+
+  filterCPUUtilizationHigh(): void {
+    const timestamp = Date.now();
+    // Allow some time to pass before we check CPU utilization.
+    if (timestamp - this.lastEventTimestamp >= this.monitoringPeriodMillis) {
+      this.lastEventTimestamp = timestamp;
+      this.observer.reduceCPUUtilization();
+    }
+  }
+
+  frameReceived(): void {
+    const timestamp = Date.now();
+    // If a enough time has passed, reset the processor and continue to monitor
+    if (timestamp - this.lastEventTimestamp >= this.monitoringPeriodMillis * 2) {
```

Comment:
So we will always ramp up `videoFramesPerFilterUpdate` after 20s by default regardless of the CPU Utilization at that time?

Reply:
This will only be called if a high cpu event did not occur in the last 10 seconds. That would mean that the utilization is under the value set in the options and we have some capacity to speed up.

Reply:
Thanks. Also can we add it to the demo also?

Reply:
Can we rename the variable `lastEventTimestamp` to indicate that it is the last timestamp from high cpu utilization event?